### PR TITLE
Added option to provide an httpAgent for proxies and other options

### LINKED ISSUE #41: Add option to specify proxy
I want to use proxy for country blocking channels

### PR DESCRIPTION
**Pull request by GilgusMaximus**
I added the httpAgent option similarly as over at yt-comment-scraper.
Basically adding the option to each exposed function and creating a new instance of the helper class each time manually via create function instead of exporting a new instantiation of the class.

Tested that everything without proxy works.
Closes #41

### CODE REVIEW

**Comment by PikachuEXE** [comment]
Is it possible to add these optional arguments as keyword arguments?
Making them as positional arguments works too.
But when the later positional argument is used,
it requires the other earlier positional arguments to be filled.

e.g.
```js
getChannelInfo(channelIdInput) // OK
getChannelInfo(channelIdInput, channelIdTypeInput) // OK
getChannelInfo(channelIdInput, channelIdType, httpAgentInput) // OK
getChannelInfo(channelIdInput, undefined, httpAgentInput) // Hmmm or do I need to put `0`?
```

**Comment by peepopoggers** [comment]
Any updates on this? Looks like a very nice change if implimented.
Pika's above comment makes alot of sense. May take a bit of time but you could use the format:
{var: val, var2: val2, var3: val3} to pass information rather than just consecutive variables otherwise it's not the most dynamic.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@PikachuEXE @ChunkyProgrammer @jasonhenriquez @peepopoggers maybe we should take over this PR and make commits to it. @GilgusMaximus will probably be unavailable till the summer https://github.com/FreeTubeApp/FreeTube/pull/1568#issuecomment-932809018

**Comment by ChunkyProgrammer** [comment]
> @PikachuEXE @ChunkyProgrammer @jasonhenriquez @peepopoggers maybe we should take over this PR and make commits to it. @GilgusMaximus will probably be unavailable till the summer [FreeTubeApp/FreeTube#1568 (comment)](https://github.com/FreeTubeApp/FreeTube/pull/1568#issuecomment-932809018)

I can look into doing this

**Comment by ChunkyProgrammer** [comment]
Should be good to review!

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
i changed the branch to dev otherwise we cant merge this

File: .eslintrc.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ module.exports = {
   // https://vuejs.github.io/eslint-plugin-vue/user-guide/#faq
   parserOptions: {
     parser: 'babel-eslint',
-    ecmaVersion: 2018,
+    ecmaVersion: 2020,
```
Comment (PikachuEXE) [inline .eslintrc.js:14]: Why change this and to this value?

File: README.md [hunk 2]
```diff
@@ -28,19 +28,24 @@ import ytch from 'yt-channel-info'
...
 
-**getChannelInfo(channelId, [channelIdType])**
-
-Returns information about a given channel ID.
-The optional argument 'channelIdType' can be provided to get faster results and less network requests if the type of channel id is known.
-- `0` = Default value used by the module. It will try all url types in the order channel -> user -> name
-- `1` = A channel id that is used with `https://www.youtube.com/channel/channelId` urls
-- `2` = A user id that is used with `https://www.youtube.com/user/channelId` urls
-- `3` = A name id that is used with `https://www.youtube.com/c/channelId` urls
+**getChannelInfo(payload)**
```
Comment (PikachuEXE) [inline README.md:31]: I am fine updating all API methods to use payload
But we need to remember to update the code on FT when we update this package

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "yt-channel-info",
-  "version": "2.2.0",
+  "version": "2.3.0",
```
Comment (PikachuEXE) [inline package.json:3]: Are we following semver for this package?
If so this should be `3.0.0` due to breaking changes

File: .eslintrc.js (hunk 1 above)
Comment (ChunkyProgrammer) [reply .eslintrc.js:14]: nullish coalescing (the ?? operator) introduced

Comment (PikachuEXE) [inline .eslintrc.js:14]:
```suggestion
    // Many methods now accept inputs in "Named parameters/keyword arguments" way
    // Which is implemented via nullish coalescing (the ?? operator) introduced in ECMAScript 2020
    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator
    ecmaVersion: 2020,
```

**Review by PikachuEXE — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (peepopoggers) [reply README.md:31]: Actually should we have a notice of this breaking change if merged?

Comment (PikachuEXE) [reply README.md:31]: > a notice of this breaking change

I don't get what you mean.

Comment (peepopoggers) [reply README.md:31]: nvm :facepalm:

File: package-lock.json [hunk 4]
```diff
@@ -1,8 +1,2090 @@
 {
   "name": "yt-channel-info",
-  "version": "2.2.0",
-  "lockfileVersion": 1,
+  "version": "3.0.0",
+  "lockfileVersion": 2,
```
Comment (PikachuEXE) [inline package-lock.json:4]: Might cause annoying diff war for people using npm 6

https://stackoverflow.com/questions/64813775/is-there-any-way-to-fix-package-lock-json-lockfileversion-so-npm-uses-a-specific

**Review by PikachuEXE — APPROVED** [review]

**Comment by peepopoggers** [comment]
Will review and test soon.

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]
LGTM!

**Review by Svallinn — APPROVED** [review]

**Pull request merged**